### PR TITLE
feat: Add printing total run duration

### LINKED ISSUE #1666: Print run duration
It would be great if Flank prints the duration of the entire execution at the end. The intermediate steps duration breakdown is really great! But it would be really nice if, at the end of _everything_, Flank could print the combined duration.

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1666

Total run duration is printed after Flank run with detailed log level set.

1. Set log level detailed
2. Run Flank
3. Check if you have time report at the end of run, ex:
```
Total run duration: 3m 11s
	- Preparation: 0m 6s
	- Running tests: 0m 6s
	- Executing matrices: 2m 50s
	- Generating reports: 0m 4s
```

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Comment by piotradamczyk5** [comment]
@flank-it

**Comment by piotradamczyk5** [comment]
> **Integration tests failed ❌**
> Windows status failure - Build scan:
> MacOS status success - Build scan: https://gradle.com/s/kppw3s6cejpa4
> Linux status success - Build scan:
> Workflow run https://github.com/Flank/flank/actions/runs/711912478

Windows failed as for others

**Comment by pawelpasterz** [comment]
@flank-it

**Comment by piotradamczyk5** [comment]
> **Integration tests failed ❌**
> Windows status failure - Build scan:
> MacOS status success - Build scan: https://gradle.com/s/jfdog3h675x2y
> Linux status success - Build scan: https://gradle.com/s/y242hhfmpwszc
> Workflow run https://github.com/Flank/flank/actions/runs/721493629

Windows failed as for others jobs

**Comment by piotradamczyk5** [comment]
@asadsalman please check if it works as expected 😃

**Comment by asadsalman** [comment]
This looks really good, thank you!

**Comment by bootstraponline** [comment]
> - Running tests: 0m 6s
> - Executing matrices: 2m 50s

what is the difference between running tests and executing matrices? Is test running a subset of the matrix execution?

File: test_runner/src/main/kotlin/ftl/Main.kt [hunk 1]
```diff
@@ -1,11 +1,14 @@
 package ftl
 
 import ftl.cli.MainCommand
+import ftl.reports.startDurationMeasurement
 import ftl.run.exception.withGlobalExceptionHandling
 import picocli.CommandLine
 
 fun main(args: Array<String>) {
+
```
Comment (Sloox) [inline test_runner/src/main/kotlin/ftl/Main.kt:9]: extra line? ;)

File: test_runner/src/main/kotlin/ftl/domain/RunTestAndroid.kt [hunk 2]
```diff
@@ -38,7 +40,7 @@ operator fun RunTestAndroid.invoke() {
     if (dryRun) {
         MockServer.start()
     }
-
+    val prepareStopWatch = StopWatch().run { start() }
```
Comment (Sloox) [inline test_runner/src/main/kotlin/ftl/domain/RunTestAndroid.kt:43]: is there a reason why it is not simply:
```suggestion
    val prepareStopWatch = StopWatch().start()
```
The start function returns a instance of this. I do not think the .run is needed here.

File: test_runner/src/test/kotlin/ftl/run/status/TestMatrixStatusPrinterTest.kt [hunk 3]
```diff
@@ -34,7 +37,7 @@ class TestMatrixStatusPrinterTest {
             every { outputStyle } returns OutputStyle.Single
         }
         val stopWatch = mockk<StopWatch>(relaxed = true) {
-            every { check(any()) } returns time
+            every { check() } returns expectedDuration
```
Comment (Sloox) [inline test_runner/src/test/kotlin/ftl/run/status/TestMatrixStatusPrinterTest.kt:40]: :)

**Review by Sloox — COMMENTED** [review]
Minor comments

**Comment by piotradamczyk5** [comment]
> Running tests: 0m 6s
> Executing matrices: 2m 50s
> what is the difference between running tests and executing matrices? Is test running a subset of the matrix execution?

Running tests stage if for preparing matrix to run (creating context, matrices, uploading files)
Executing matrices is running them on the FTL side.
Test running is not a subset of matrix execution, naming for those stages are inspired by code methods

File: test_runner/src/main/kotlin/ftl/domain/RunTestAndroid.kt (hunk 2 above)
Comment (piotradamczyk5) [reply test_runner/src/main/kotlin/ftl/domain/RunTestAndroid.kt:43]: my mistake, fixed. Thanks!

File: test_runner/src/main/kotlin/ftl/Main.kt (hunk 1 above)
Comment (piotradamczyk5) [reply test_runner/src/main/kotlin/ftl/Main.kt:9]: removed

**Review by Sloox — APPROVED** [review]
LGTM

**Comment by Sloox** [comment]
@flank-it

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**